kolibri/core/logger/test/test_integrated_api.py

```diff
@@ -2676,5 +2731,148 @@ def test_update_assessment_session_update_attempt_in_lesson_succeeds(self):
+        session_log = ContentSessionLog.objects.create(
+            user=None,
+            content_id=self.content_id,
+            channel_id=self.channel_id,
+            start_timestamp=local_now(),
+            end_timestamp=local_now(),
+            kind="exercise",
+            extra_fields={"context": {"node_id": self.node.id, "mastery_level": -1}},
+        )
+        AttemptLog.objects.create(
+            masterylog=None,
+            sessionlog=session_log,
+            start_timestamp=timestamp,
+            end_timestamp=timestamp,
+            correct=0,
+            item=self.item,
+            user=None,
+            interaction_history=[],
+        )
+        # Update self.session_log to ensure _make_request works as expected.
+        self.session_log = ContentSessionLog.objects.create(
+            user=None,
+            content_id=self.content_id,
+            channel_id=self.channel_id,
+            start_timestamp=local_now(),
+            end_timestamp=local_now(),
+            kind="exercise",
+            extra_fields={"context": {"node_id": self.node.id, "mastery_level": -1}},
+        )
+        hinteraction = {
+            "type": interaction_types.HINT,
+            "answer": {"response": "hinty mchintyson"},
+        }
+        attemptlog = AttemptLog.objects.create(
+            masterylog=None,
+            sessionlog=self.session_log,
+            start_timestamp=timestamp,
+            end_timestamp=timestamp,
+            correct=0,
+            item=self.item,
+            user=None,
+            interaction_history=[hinteraction],
+        )
+        response = self._make_request(
+            {
+                "interactions": [
+                    {
+                        "item": self.item,
+                        "answer": {"response": "hinty mchintyson2"},
+                        "hinted": True,
+                        "correct": 0,
+                        "time_spent": 10,
+                        "replace": True,
+                    }
+                ],
+            }
+        )
+
+        self.assertEqual(response.status_code, 200)
+        attempt_id = response.json().get("attempts", [{}])[0].get("id")
+        self.assertIsNotNone(attempt_id)
+        self.assertEqual(attemptlog.id, attempt_id)
+        try:
+            attempt = AttemptLog.objects.get(id=attempt_id)
+        except AttemptLog.DoesNotExist:
+            self.fail("Nonexistent attempt_id returned")
+        self.assertEqual(attempt.correct, 0)
+        self.assertEqual(attempt.answer, {"response": "hinty mchintyson2"})
+        self.assertEqual(attempt.time_spent, 10)
+        self.assertEqual(attempt.interaction_history[0], hinteraction)
+        self.assertEqual(
+            attempt.interaction_history[1],
+            {
+                "type": interaction_types.HINT,
+                "answer": {"response": "hinty mchintyson2"},
+            },
+        )
```

Comment:
This may be covered and I missed it, but: also good to do a query to make sure no other attemptlog exists for that item/user, to cover all the bases?

Reply:
For the non-anonymous cases, yes, that would be a good idea.

Reply:
Have added these assertions.